app/controllers/theft_alerts_controller.rb

```diff
@@ -20,13 +20,15 @@ def create
   rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotFound => e
```

Comment:
Is it possible to split these different actions into different controller actions?

Reply:
If by actions you mean (1) creating the theft alert and (2) creating a payment: We can't redirect a POST request, but I'm open to hearing suggestions. Payment creation could and definitely should go in a background job.

Reply:
That is what I mean by 2 actions. I imagine, with the current flow, that you would create (or update an existing inactive) theft alert before rendering the `alert_purchase` template.

w/r/t background processing of payments - just running through this with stripes test cards for failures, it seems like stripe manages all the payment verification before actually sending us a response (e.g. if the card is declined, it shows it in the form before reloading the page). So I'm not sure what backgrounding the payment creation would get us?

Reply:
After seeing how smooth stripe's process is for handling problems with payments now, I'm not concerned about splitting this into multiple different actions.

This little article I think covered my opinions about using the bang method for save (though I don't think the bang method should be removed, there are many uses for it) - https://thoughtbot.com/blog/save-bang-your-head-active-record-will-drive-you-mad - I think the `save!` in the theft controller is because theft alerts aren't using a traditional active record validation?

I still think it would be great to get the theft alert created before making the payment, in a separate action.

But **this isn't high priority**, I assume we're going to extend this process with additional steps for writing text in the alert, etc, and we can do the refactoring then.